cgi-bin/samygo-web-api.cgi

```diff
@@ -270,68 +270,76 @@ $challenge = trim(file_get_contents("/mnt/etc/samygo-web-api.challenge"));
+				if(getLock()){
+					$exec=`$SAMYGOSO -d -A -B -l "$LIBSDIR/libLogChannel.so" 2>/dev/null 1>&2`;
+					releaseLock();
+					sleep(1);
 				
 				
-				#process the log
-				$channel_info=file_get_contents("/dtv/LogChannel.log");
+					#process the log
+					$channel_info=file_get_contents("/dtv/LogChannel.log");
 				
-				#$channel_info=`$BIN/cat /dtv/LogChannel.log | $BIN/grep "Source:" -A 8 | $BIN/sed -e "s/[^ ]* \(.*\)/\1/g"`;
-				error_log("Received output: $channel_info");
-				//Split the output into tokens:
-				/*
-	Source: TV (0)
-	PVR Status: NONE
-	PowerState: Normal
-	TV mode: Cable (1)
-	Volume: 6 (mute)
-	Channel number: 44
-	Channel name: Nat Geo
-	Program name: Geniul: Einstein
-	Resolution: 720x576
-				 */
-				 $channel_lines=explode("\n", $channel_info);
-				 $output = array();
-				 foreach ($channel_lines as $line){
-					 error_log("Looking at $line");
-					 if(preg_match('/\[[^\]]+\]\s+([^:]+):(.*)/', $line, $tokens)){
+					#$channel_info=`$BIN/cat /dtv/LogChannel.log | $BIN/grep "Source:" -A 8 | $BIN/sed -e "s/[^ ]* \(.*\)/\1/g"`;
+					error_log("Received output: $channel_info");
+					//Split the output into tokens:
+					/*
+						Source: TV (0)
+						PVR Status: NONE
+						PowerState: Normal
+						TV mode: Cable (1)
+						Volume: 6 (mute)
+						Channel number: 44
+						Channel name: Nat Geo
+						Program name: Geniul: Einstein
+						Resolution: 720x576
+					*/
+					$channel_lines=explode("\n", $channel_info);
+					$output = array();
+					foreach ($channel_lines as $line){
+						error_log("Looking at $line");
+						if(preg_match('/\[[^\]]+\]\s+([^:]+):(.*)/', $line, $tokens)){
 						 
-						 #$tokens = explode(":", $line);
+							#$tokens = explode(":", $line);
 						 
-						 #process the first token to remove spaces, lowercase
-						 $tokens[1]=preg_replace('/\s+/', '_', $tokens[1]);
-						 $tokens[1]=strtolower($tokens[1]);
-						 $tokens[2]=trim($tokens[2]);
+							#process the first token to remove spaces, lowercase
+							$tokens[1]=preg_replace('/\s+/', '_', $tokens[1]);
+							$tokens[1]=strtolower($tokens[1]);
+							$tokens[2]=trim($tokens[2]);
 						 
-						 #save it into the output
-						 $output[$tokens[1]]=$tokens[2];
+							#save it into the output
+							$output[$tokens[1]]=$tokens[2];
 						 
-						 #if power_state == Soft-Off don't record the program name
-						 if($tokens[1] == 'program_name' && $output['power_state'] == 'Soft-Off'){
-							 error_log("The TV is off. Ignoring program name");
-							 $output[$tokens[1]] = 'Off';
-						 }
-						 
-						 if($tokens[1] == 'program_name' && preg_match('/HDMI|Undefined/', $output['source'])){
-							 error_log("The TV is tuned to a different source.");
-							 $output[$tokens[1]] = 'Non-TV';
-						 }
-						 error_log("Saving $tokens[1] = $tokens[2]");
-					}
+							#if power_state == Soft-Off don't record the program name
+							if($tokens[1] == 'program_name' && $output['power_state'] == 'Soft-Off'){
```

Comment:
This check can maybe be removed as well after this patch, although it will require people to have /dtv/tvIsSleeping set up for CHANNELINFO to work correctly, even if their TV is reporting soft-off.

Reply:
The purpose of this is to prevent a system like Home Assistant which periodically polls the TV status from having to store all the TV program names that go on when the TV is off. So the API doesn't report the program name unless the power is on.

Reply:
The point of this patch is to return an empty array when the TV is in soft-off mode. My TV does not report 'Soft-Off' in $output['power_state'], so I have to rely on /dtv/tvIsSleeping.
After this patch, if /dtv/tvIsSleeping exists, it will return an empty array. If the tv is 'hard-off' it will not respond anyway.
As I see it, the only reason to consider keeping this check is for people that have a tv which supports reporting soft-off through libLogChannel AND have soft-off configured without the power on/off commands that take care of /dtv/tvIsSleeping.